controller/src/job/mod.rs

```diff
@@ -206,15 +206,16 @@ pub(crate) async fn archive_logs(k8s_client: kube::Client, job_name: &str) -> Jo
         SystemTime::now().duration_since(UNIX_EPOCH)?.as_secs()
     );
 
-    client
+    if client
         .create_log_stream()
         .log_group_name("testsys")
         .log_stream_name(&name)
         .send()
         .await
-        .context(error::CreateLogStreamSnafu {
-            log_stream: name.to_string(),
-        })?;
+        .is_err()
+    {
+        trace!("Unable to create log group it is possible it already exists.");
```

Comment:
A couple of things, you should log the error with `{:?}`.
You should check for the specific "it already exists" error condition and only ignore that, otherwise issues like authorization, etc could be difficult to diagnose.